libraries/AP_Beacon/AP_Beacon_Pozyx.cpp

```diff
@@ -0,0 +1,175 @@
+    AP_Beacon_Backend(frontend),
+    linebuf_len(0)
+{
+    uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Beacon, 0);
+    if (uart != nullptr) {
+        uart->begin(serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_Beacon, 0));
+    }
+}
+
+// return true if sensor is basically healthy (we are receiving data)
+bool AP_Beacon_Pozyx::healthy()
+{
+    // healthy if we have parsed a message within the past 300ms
+    return ((AP_HAL::millis() - last_update_ms) < AP_BEACON_TIMEOUT_MS);
+}
+
+// update the state of the sensor
+void AP_Beacon_Pozyx::update(void)
+{
+    static uint8_t counter = 0;
+    counter++;
+    if (counter > 200) {
+        counter = 0;
+        //::printf("b\n");
+    }
+
+    if (uart == nullptr) {
+        return;
+    }
+
+    // read any available characters
+    int16_t nbytes = uart->available();
+    while (nbytes-- > 0) {
+        char c = uart->read();
+
+        switch (parse_state) {
+
+            default:
+            case ParseState_WaitingForHeader:
+                if (c == AP_BEACON_POZYX_HEADER) {
+                    parse_state = ParseState_WaitingForMsgId;
+                    linebuf_len = 0;
+                }
+                break;
+
+            case ParseState_WaitingForMsgId:
+                parse_msg_id = c;
+                if (parse_msg_id == AP_BEACON_POZYX_MSGID_BEACON_CONFIG ||
+                    parse_msg_id == AP_BEACON_POZYX_MSGID_BEACON_DIST ||
+                    parse_msg_id == AP_BEACON_POZYX_MSGID_POSITION) {
+                    parse_state = ParseState_WaitingForLen;
+                } else {
+                    // invalid message id
+                    parse_state = ParseState_WaitingForHeader;
+                }
+                break;
+
+            case ParseState_WaitingForLen:
+                parse_msg_len = c;
+                if (parse_msg_len > AP_BEACON_POZYX_MSG_LEN_MAX) {
+                    // invalid message length
+                    parse_state = ParseState_WaitingForHeader;
+                } else {
+                    parse_state = ParseState_WaitingForContents;
+                }
+                break;
+
+            case ParseState_WaitingForContents:
+                // add to buffer
+                linebuf[linebuf_len++] = c;
+                if ((linebuf_len == parse_msg_len) || (linebuf_len == sizeof(linebuf))) {
+                    // process buffer
+                    parse_buffer();
+                    // reset state for next message
+                    parse_state = ParseState_WaitingForHeader;
+                }
+                break;
+        }
+    }
+}
+
+// parse buffer
+void AP_Beacon_Pozyx::parse_buffer()
+{
+    // check crc
+    uint8_t checksum = 0;
+    checksum ^= parse_msg_id;
+    checksum ^= parse_msg_len;
+    for (uint8_t i=0; i<linebuf_len-1; i++) {
+        checksum ^= linebuf[i];
+    }
+    // return if failed checksum check
+    if (checksum != linebuf[linebuf_len-1]) {
```

Comment:
When exclusive OR is performed up to the calculation result, it becomes 0.
Therefore,
I think that it should be judged by whether the value of cecksum is 0 or not.